flyway-core/src/main/java/org/flywaydb/core/internal/util/scanner/classpath/ClassPathScanner.java

```diff
@@ -199,16 +199,23 @@ private String toClassName(String resourceName) {
                     }
 
                     try {
-                        boolean directoryFound = false;
-                        Enumeration<JarEntry> entries = jarFile.entries();
-                        while (entries.hasMoreElements()) {
-                            if (entries.nextElement().isDirectory()) {
-                                directoryFound = true;
-                                break;
+                        // Look up resources from jar files within an Ant task
+                        boolean shouldLookupResources = classLoader.getParent() != null
+                                            && "AntClassLoader".equals(classLoader.getParent().getClass().getSimpleName());
```

Comment:
Flyway Ant Tasks cannot lookup Java or SQL based migrations.  If my migrations come in different artifacts (as JARs) it would be nice when the Ant Task can take care of it and lookup the migrations from those JARs as well. So far I need to unpack the content of JARs into the classpath folder which brings more work on the build side and breaks the structure of the migrations (for example, my JARs pack different types of migrations in different namespaces).
This cannot be done by the AntClassLoader, the default class loader of Ant. So I used URLClassLoader and packed the AntClassLoader as its parent.